kpm/lib/kpm/base_artifact.rb

```diff
@@ -164,8 +164,10 @@ def skip_if_exists(artifact_info, coordinates, sha1_file)
         end
 
         # Finally check if remote_sha1 matches what we have locally
-        local_sha1 = Digest::SHA1.file(artifact_info[:file_path]).hexdigest
-        local_sha1 == artifact_info[:sha1]
+        if !artifact_info[:is_tgz]
+          local_sha1 = Digest::SHA1.file(artifact_info[:file_path]).hexdigest
+          local_sha1 == artifact_info[:sha1]
+        end
```

Comment:
Actually, I don't think this is needed anymore (the tgz check, that is). I was seeing issues where we'd try to run the sha1 computation for a folder. But this was actually caused by a bad spec that wasn't using the right class to pull a plugin.